apps/desktop-wallet/src/api/apiDataHooks/transaction/useFetchTransaction.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { useQuery } from '@tanstack/react-query'
+
+import { confirmedTransactionQuery, pendingTransactionQuery } from '@/api/queries/transactionQueries'
+import { selectSentTransactionByHash } from '@/features/sentTransactions/sentTransactionsSelectors'
+import { useAppSelector } from '@/hooks/redux'
+
+interface UseFetchTransactionProps {
+  txHash: string
+}
+
+const useFetchTransaction = ({ txHash }: UseFetchTransactionProps) => {
+  const sentTx = useAppSelector((s) => selectSentTransactionByHash(s, txHash))
+
+  const isPendingTx = sentTx && sentTx.status !== 'confirmed'
+
+  const { data: confirmedTx, isLoading: isLoadingConfirmedTx } = useQuery(
+    confirmedTransactionQuery({ txHash, skip: isPendingTx })
+  )
+  const { data: pendingTx, isLoading: isLoadingPendingTx } = useQuery(
+    pendingTransactionQuery({ txHash, skip: !isPendingTx })
+  )
```

Comment:
Both of these are using the same endpoint, but with different tanstack config. The first one as `staleTime: Infinity` since its data will never change, and the second one has `refetchInterval: 3000`